tests/unit_tests/ModuleBindingTests.cpp

```diff
@@ -158,4 +158,37 @@ endmodule
     CHECK(foo.value.integer() == 1);
 }
 
+TEST_CASE("Module children (loop generate)", "[binding:modules]") {
```

Comment:
Do we also have / we should have tests to make sure that diagnostics do reasonable things when you have a loop generate and the expressions aren't actually constant and fail to evaluate?

Change looks good to me in general